### ISSUE TITLE
Automaticaly enable for newly created users (if they have an email address)

### ISSUE THREAD

**Issue by waylon2002my** [nursoda/twofactor_email#83]
Hi Guys, I would like to ask is that possible for new user to enable automatic email authentication once the administrator created the account?

thanks.

**Comment by waylon2002my** [comment]
may i know this feature will be approve by design team?

**Comment by towerplease** [comment]
Hey there, 
as @waylon2002my already requested, it would be supernice if the e-mail authentication would be pre-enabled when i create a user. We starting to implement Nextcloud here and we need to sit by every user and enable it manually.

Would be nice!

Thank you!

**Comment by axheli** [comment]
Hello, are we still waiting for this to be implemented?

**Comment by axheli** [comment]
@rullzer i just enabled this feature and under administration > Security > 2FA is checked like below image but once the user tries to login they get below image. 
![Screen Shot 2021-01-18 at 10 09 19 AM](https://user-images.githubusercontent.com/59488153/104939373-f74d0e00-5975-11eb-9079-97aa30b09d3c.png)

The  way i got around it was that i downloaded Two-Factor Admin support app generated a key logged in as the user then i went under Settings > Personal > Security > and i had to manual enable the Email verification then i got a token via email added the token then i was able to login as the user and 2FA via email works. 

My question is once the admin enables the 2fa is there a way to have it as once the user logins in for the first time they can setup and verify in the below location. 
![Screen Shot 2021-01-18 at 10 09 19 AM](https://user-images.githubusercontent.com/59488153/104940198-054f5e80-5977-11eb-9b0c-8ff663dedb6d.png)

**Comment by rullzer** [comment]
Yes we now have a mechanism to allow setting up 2FA on login (if it is enforced).
However, I ahve not have much time to spend on this app.
BUt if somebody provides a PR i'd be happy to have a look.

**Comment by andyxheli** [comment]
@Simounet Im not sure how familiar you're with 2fa but wanted to see if you'd be interested to help with this ☺️

**Comment by Simounet** [comment]
@andyxheli yes I am but not with email. I'm sorry but I have a few other fixes to do to NC before spending time on this one.

**Comment by andyxheli** [comment]
@Simounet no problem at all thank you so much for taking your time to reply.

**Comment by andyxheli** [comment]
@nursoda What do you think about this and would it be complicated to implement

**Comment by nursoda** [comment]
Can't tell yet. I'll review all bugs. But nor now I have other priorities, like fixing new dev dependency / security issues, pimping the app's appearance in the app store, etc.

**Comment by andyxheli** [comment]
> Can't tell yet. I'll review all bugs. But nor now I have other priorities, like fixing new dev dependency / security issues, pimping the app's appearance in the app store, etc.

Sounds good. thank you for taking your time to respond.

**Comment by nursoda** [comment]
I simply don't know (yet) how to trigger code from this app from the process of "setting up a new user". I have too little knowledge of the whole NC architecture (yet), so any input on this (routing? another app implementing a similar trigger) would be helpful. Or does it need to be implemented in 'server'? I'd think so! IF that were the case, this bug would need to be reported [there](https://github.com/nextcloud/server/issues).

**Comment by andyxheli** [comment]
The TOTP app currently dose it if a user doesn't have 2FA enabled but it's required by server admin it will ask you to setup it up upon login. If you make 2FA required and only have 2FA via email it will not ask you to set it up for upon login. Dose that make sense.

**Comment by nursoda** [comment]
Well, to be honest, I consider twofactor_email much less secure than TOTP or U2F. That said, it's the thing that an admin really CAN enforce (if he/she is able to set an email address, which I consider to be the case in many scenarios). So, as admin, I'd not ONLY offer twofactor_email, but I'd offer it as last resort – and maybe I'd even enable it as default to be able to enforce 2FA. But that would require a method to enable it for EXISTING users also (which is not scope of THIS issue).

Your point was that one might enable it if ONLY twofactor_email is present. I sincerely hope that this is a rare case that I don't want to foster. (@folks, please use proper 2FA :) )

Nevertheless, independent of whether there's only twofactor_email or other twofactor_* apps present, one could want to AT LEAST use twofactor_email if no other means is enabled yet (and an email address is set).

Depending on whether we use the primary(→¹) (or a secondary?) notification address (set in /settings/user) or use an separate(→²) address that has to be set up in /settings/user/security/twofactor_email (MAY it be the same as the primary address?), we ¹could or ²could not set up twofactor_email for new/existing users WITHOUT interaction upon an admin's request.

**Comment by thomas-mc-work** [comment]
> Plus, an admin can use OCC to bulk-enable 2FA for a group of users.

@nursoda Could you please tell me how this can be done?

**Comment by nursoda** [comment]
I wrote this soon after I started to maintain the app. I don't know what I had in mind. Probably, that one can enable/disable 2FA altogether for a user.

Reviewing my post and trying to do what I suggested is possible, I was not able to do it myself. So my post is wrong in that respect. I think that I need to create an enhancement issue that allows an admin to enable twofactor_email for existing users. At least via OCC (in the twofactorauth and/or twofactor_email namespace, ideally also via web interface.

Currently, there's only this occ command:

```
$ occ twofactorauth:disable USER email
The provider does not support this operation.
$ occ twofactorauth:enable USER email
The provider does not support this operation.
```
